cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -165,7 +165,7 @@ func (k *proxy) NewClient() (client.Client, error) {
 	var c client.Client
 	// Nb. The operation is wrapped in a retry loop to make newClientSet more resilient to temporary connection problems.
 	connectBackoff := newConnectBackoff()
-	if err := retryWithExponentialBackoff(connectBackoff, func() error {
+	if err := retryWithExponentialBackoff(context.Background(), connectBackoff, func(_ context.Context) error {
```

Comment:
Can we open a follow-up issue to make sure also all the context.Background() introduced by this PR are removed?

Reply:
If we have to add temporary context's that should be eventually replaced later, please always add context.TODO(). This is the reason why context.TODO() exists (emphasize on the `TODO` part :))